### ISSUE TITLE
Supermaven keeps stopping in neovim

### ISSUE THREAD

**Issue by MDSADABWASIM** [supermaven-inc/supermaven-nvim#87]
Hey, I'm a supermaven pro user, and I'm constantly facing this issue, it keeps stopping, even after running `:SupermavenRestart` it didn't work in my react-native project, if I switch to my config files(Lua), it's working fine there which is quite strange.

Here's my config:
```
  {
    "supermaven-inc/supermaven-nvim",
    config = function()
      require("supermaven-nvim").setup({
        keymaps = {
          accept_suggestion = "<C-y>",
          clear_suggestion = "<C-]>",
          accept_word = "<C-j>",
        },
        log_level = "info",                -- set to "off" to disable logging completely
        disable_inline_completion = false, -- disables inline completion for use with cmp
        disable_keymaps = false            -- disables built in keymaps for more manual control
      })
    end,
  }
```

**Comment by sm-victorw** [comment]
Could you create an `sm-log.txt` file in your home directory, and see if there is anything that suggests an error? It sounds like it could be the binary running into some issue when it tries to index the repository

**Comment by MDSADABWASIM** [comment]
Here's the log file.

**Comment by sm-victorw** [comment]
We're going to change the plugin to be compatible with a new version of the sm-agent binary which shouldn't run into errors like this - though until this is done I don't think there is any temporary fix available

**Comment by MDSADABWASIM** [comment]
Ok @sm-victorw please update me on this thread when it's done, thanks for the quick reply though.

**Comment by sm-victorw** [comment]
Sorry for the long delay @MDSADABWASIM , the branch `rust-binary` contains code that I think will eventually be merged into `main`, after some more testing. If I am correct in that the issue you were running into was a result of the old `sm-agent` binary, using this branch should work for you.

Using e.g. the Lazy plugin manager you could set `branch = 'rust-binary',` if you want to try it out

**Comment by sm-victorw** [comment]
I believe this should be solved with #109, if the problem persists please reopen the issue

**Comment by celsobenedetti** [comment]
As of today, the plugin agent keeps stopping for me. Cannot use the plugin reliably:


```
[TRACE  Sun 12 Jan 2025 04:41:39 PM -03] Starting Supermaven...
[DEBUG  Sun 12 Jan 2025 04:41:39 PM -03] Unknown message: 2025-01-12T19:41:39.847Z INFO  [sm_agent] Starting sm-agent
[TRACE  Sun 12 Jan 2025 04:41:39 PM -03] Stopping Supermaven...
[TRACE  Sun 12 Jan 2025 04:43:20 PM -03] Starting Supermaven...
[DEBUG  Sun 12 Jan 2025 04:43:20 PM -03] Unknown message: 2025-01-12T19:43:20.972Z INFO  [sm_agent] Starting sm-agent
[TRACE  Sun 12 Jan 2025 04:43:21 PM -03] Stopping Supermaven...
```


#### Info

```
branch: main
commit: 07d20fc
```

---

```
OS: Pop!_OS 22.04 LTS x86_64
Host: MS-7D75 1.0
Kernel: 6.9.3-76060903-generic
CPU: AMD Ryzen 7 7800X3D (16) @ 5.050GHz
```
